packages/aws-cdk-lib/aws-sns-subscriptions/test/firehose.test.ts

```diff
@@ -0,0 +1,334 @@
+      TopicArn: { Ref: 'MyTopic86869434' },
+    });
+    Template.fromStack(stack).hasResourceProperties('AWS::SQS::QueuePolicy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'sqs:SendMessage',
+            Condition: {
+              ArnEquals: {
+                'aws:SourceArn': { Ref: 'MyTopic86869434' },
+              },
+            },
+            Effect: 'Allow',
+            Principal: { Service: 'sns.amazonaws.com' },
+            Resource: { 'Fn::GetAtt': ['DeadLetterQueue9F481546', 'Arn'] },
+          },
+        ],
+      },
+      Queues: [{ Ref: 'DeadLetterQueue9F481546' }],
+    });
+  });
+});
+
+describe('Amazon Data Firehose delivery stream subscription cross-stack', () => {
+  test('creates cross-region configuration', () => {
+    // GIVEN
+    const app = new App();
+    const topicStack = new Stack(app, 'TopicStack', { env: { account: '11111111111', region: 'us-east-1' } });
+    const firehoseStack = new Stack(app, 'FirehoseStack', { env: { account: '11111111111', region: 'us-east-2' } });
+
+    const topic = new sns.Topic(topicStack, 'Topic', {
+      displayName: 'displayName',
+      topicName: 'topicName',
+    });
+
+    const bucket = new s3.Bucket(firehoseStack, 'Bucket');
+    const deliveryStream = new firehose.DeliveryStream(firehoseStack, 'DeliveryStream', {
+      destination: new firehose.S3Bucket(bucket),
+    });
+
+    // WHEN
+    topic.addSubscription(new subs.FirehoseSubscription(deliveryStream));
+
+    // THEN
+    Template.fromStack(topicStack).hasResourceProperties('AWS::SNS::Topic', {
+      DisplayName: 'displayName',
+      TopicName: 'topicName',
+    });
+    Template.fromStack(firehoseStack).hasResourceProperties('AWS::SNS::Subscription', {
+      Endpoint: { 'Fn::GetAtt': ['DeliveryStream58CF96DB', 'Arn'] },
+      Protocol: 'firehose',
+      Region: 'us-east-1',
+      SubscriptionRoleArn: { 'Fn::GetAtt': ['DeliveryStreamTopicSubscriptionRole4964AFE6', 'Arn'] },
+      TopicArn: { 'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, ':sns:us-east-1:11111111111:topicName']] },
+    });
+    Template.fromStack(firehoseStack).hasResourceProperties('AWS::IAM::Role', {
+      AssumeRolePolicyDocument: {
+        Statement: [{
+          Action: 'sts:AssumeRole',
+          Effect: 'Allow',
+          Principal: { Service: 'sns.amazonaws.com' },
+        }],
+      },
+    });
+    Template.fromStack(firehoseStack).hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [{
+          Action: [
+            'firehose:DescribeDeliveryStream',
+            'firehose:ListDeliveryStreams',
+            'firehose:ListTagsForDeliveryStream',
+            'firehose:PutRecord',
+            'firehose:PutRecordBatch',
+          ],
+          Effect: 'Allow',
+          Resource: { 'Fn::GetAtt': ['DeliveryStream58CF96DB', 'Arn'] },
+        }],
+      },
+    });
```

Comment:
You can add a check to make sure that stack dependencies are set correctly. Check that `firehoseStack.dependencies()` includes `topicStack`.